### PR TITLE
Switch to ocis-accounts as the user backend

### PR DESCRIPTION
**Pull request by butonic**
To get access to guest accounts we will use the ocis-accounts service instead of an ldap server

part of https://github.com/owncloud/ocis-glauth/issues/13

- killed groups ... might come back when accounts supports them

- [x] implement basic auth

### CODE REVIEW

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/graph.go  3
- pkg/server/http/server.go  1
         

Complexity decreasing per file
==============================
+ pkg/service/v0/users.go  -1
         

Clones removed
==============
+ pkg/service/v0/users.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-graph/pullRequest?prid=5488655&bid=17987088)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/graph.go  3
- pkg/server/http/server.go  1
         

Complexity decreasing per file
==============================
+ pkg/service/v0/users.go  -1
         

Clones removed
==============
+ pkg/service/v0/users.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-graph/pullRequest?prid=5488655&bid=17987088)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/middleware/basicauth.go  3
- pkg/middleware/option.go  2
- pkg/service/v0/graph.go  3
- pkg/server/http/server.go  1
         

Complexity decreasing per file
==============================
+ pkg/service/v0/users.go  -1
         

Clones added
============
- pkg/service/v0/option.go  1
- pkg/middleware/option.go  2
         

Clones removed
==============
+ pkg/service/v0/users.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-graph/pullRequest?prid=5488655&bid=17987088)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/middleware/option.go  2
- pkg/server/http/server.go  1
- pkg/middleware/basicauth.go  3
- pkg/service/v0/users.go  3
         

Clones added
============
- pkg/service/v0/option.go  1
- pkg/middleware/option.go  2
         

Clones removed
==============
+ pkg/service/v0/users.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-graph/pullRequest?prid=5488655&bid=17987088)

**Comment by butonic** [comment]
~~this PR kills the bridge setup because it no longer uses LDAP as the backend but ocis-accounts~~

~~possible ways forward:~~
~~- implement an ocis-graph-ldap that uses ldap as the backend?~~
~~- implement a new ocis-accounts-ldap that does not persist accounts itself but instead acts as a protocol translator to a real ldap server~~
~~- do not directly talk to the accounts service but use ocis-glauth, which [will remain read only for now](https://github.com/owncloud/ocis-accounts/pull/30#issuecomment-644152756)~~

the bridge uses konnectd, which talks to ocis-glauth (with the owncloud backend) which talks to owncloud10 with the graph app. the recent changes would make konnectd talk to ocis-glauth with ocis-accounts as the backend directly. ocis-proxy would authenticate the user and look up the id in ocis-accounts. if the user is not migrated requests will be forwarded to oc10 with the openidconnect app (or we teach it the internal jwt token). ocis-graph is not necessary there, yet.

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/server/http/server.go  1
- pkg/middleware/option.go  2
- pkg/service/v0/users.go  3
- pkg/middleware/basicauth.go  3
         

Clones added
============
- pkg/service/v0/option.go  1
- pkg/middleware/option.go  2
         

Clones removed
==============
+ pkg/service/v0/users.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-graph/pullRequest?prid=5488655&bid=17987088)

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/service/v0/users.go  3
- pkg/server/http/server.go  1
- pkg/middleware/basicauth.go  3
- pkg/middleware/option.go  2
         

Clones added
============
- pkg/service/v0/option.go  1
- pkg/middleware/option.go  2
         

Clones removed
==============
+ pkg/service/v0/users.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-graph/pullRequest?prid=5488655&bid=17987088)

File: pkg/flagset/flagset.go [hunk 1]
```diff
@@ -202,7 +160,7 @@ func ServerWithConfig(cfg *config.Config) []cli.Flag {
 			Name:        "reva-gateway-addr",
 			Value:       "127.0.0.1:9142",
 			Usage:       "REVA Gateway Endpoint",
-			EnvVars:     []string{"REVA_GATEWAY_ADDR"},
+			EnvVars:     []string{"GRAPH_REVA_GATEWAY_ADDR"},
```
Comment (refs) [inline pkg/flagset/flagset.go:163]: why renaming this when it actually refers to the gateway `127.0.0.1:9142`?

File: pkg/server/http/server.go [hunk 2]
```diff
@@ -22,7 +26,13 @@ func Server(opts ...Option) (http.Service, error) {
 		http.Flags(options.Flags...),
 	)
 
+	as, err := getAccountsService()
+	if err != nil {
+		return service, err
```
Comment (refs) [inline pkg/server/http/server.go:31]: shouldn't it be `return nil, err`? Or are we actually ignoring this error?

File: pkg/service/v0/users.go [hunk 3]
```diff
@@ -4,87 +4,142 @@ import (
...
 
 // UserCtx middleware is used to load an User object from
 // the URL parameters passed through as the request. In case
 // the User could not be found, we stop here and return a 404.
 func (g Graph) UserCtx(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		var user *ldap.Entry
 		var err error
 
 		userID := chi.URLParam(r, "userID")
```
Comment (refs) [inline pkg/service/v0/users.go:27]: I wish we use Chi on Reva 👏

**Review by refs — APPROVED** [review]
Couldn't try the logic, but code looks good

**Pull request closed without merging**